brave/brave-core
# Terms of service update rewards panel

Resolves https://github.com/brave/brave-browser/issues/35978



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

### Test Steps
#### Test 1:
  * With a new profile, enable Rewards.
  * close the app.
  * Increment the "tosVersion" field of the Rewards parameters endpoint.
  * Restart the browser.
  * Verify that the ToS update notice is displayed on the Rewards panel.
  * Verify that clicking the Terms of Service link opens a new tab with the correct URL.
  * Verify that clicking the "reset" link shows the dialog to relaunch app and check if rewards state is reset.
#### Test 2:
  * With a new profile, enable Rewards.
  * close the app.
  * Increment the "tosVersion" field of the Rewards parameters endpoint.
  * Restart the browser.
  * Verify that the ToS update notice is displayed on the Rewards panel.
  * Click the `i accept` button.
  * Verify that the ToS update notice is no longer displayed.

### Upgrade Case

If the "tosVersion" field returned from the "parameters" endpoint has a value of `1`, then the ToS update notice should not be displayed when the user upgrades their browser.

* With a profile created in a previous version, with Rewards enabled:
* Ensure that the "parameters" endpoint is returning `1` for the "tosVersion" field.
* Open the browser.
* Verify that the ToS update notice is not displayed.
